WalletWasabi/Affiliation/Serialization/AffiliationFlagJsonConverter.cs

```diff
@@ -0,0 +1,30 @@
+using Newtonsoft.Json;
+
+namespace WalletWasabi.Affiliation.Serialization;
+
+public class AffiliationFlagJsonConverter : JsonConverter<AffiliationFlag>
+{
+	public override AffiliationFlag? ReadJson(JsonReader reader, Type objectType, AffiliationFlag? existingValue, bool hasExistingValue, JsonSerializer serializer)
+	{
+		if (reader.Value is string serialized)
+		{
+			return new AffiliationFlag(serialized);
+		}
+		else
+		{
+			throw new Exception();
```

Comment:
> Btw, in most of the cases we return null here because that's the expected behavior according to the docs and the method signature (AffiliationFlag?)

I want the serializer to be strict and don't allow it to return `null`. I concluded `JsonSerializationException` is the most appropriate.  See https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/4408ad25fb9579305f1c8a27b61f2be6e22b0b24.